testcafe/pages/ServicePage.js

```diff
@@ -0,0 +1,22 @@
+import { ReactSelector } from 'testcafe-react-selectors';
+import config from '../config';
+
+export default class ResourcePage {
+  constructor() {
+    const baseSelectorName = 'ServicePage';
+    const baseSelector = ReactSelector(baseSelectorName);
+    this.serviceName = baseSelector.find('.listing--main--left header');
+    this.description = baseSelector.find('.listing--main--left--about');
+    this.details = baseSelector.find('.listing--main--left--details');
+    // TODO: Can't use nested React component name PhoneNumber because it is
+    // instantiated in both the header and the body of the page and because the
+    // testcafe-react-selectors plugin is currently unable to mix CSS selectors
+    // in between React component names.
+    // https://github.com/DevExpress/testcafe-react-selectors/issues/51
```

Comment:
That ticket is out of date. You can now nest both CSS and React selectors under other ReactSelectors with the `.findReact()` method: https://github.com/DevExpress/testcafe-react-selectors#searching-for-nested-components

This also means that you don't need to add all the CSS classes that you added just for the sake of selecting components for tests.

Reply:
The problem I have with that is then the tests are dependent on the layout of the HTML; by relying on classes, those elements can move anywhere on the page and the tests won't break. For example, the line below (editButton) would turn into `this.editButton = baseSelector.findReact('ListPageSidebar').findReact('ul').nth(0)` if I didn't have the class.

Reply:
Okay, yeah, that's a fair comment, though I feel like our CSS names are a bit too generic and there's a minor risk of accidentally selecting something else. But we can deal with that later.